### ISSUE TITLE
Code sample in documentation causes exception

### ISSUE THREAD

**Issue by Dotrar** [py-pdf/pypdf#2050]
Reading [this](https://pypdf2.readthedocs.io/en/latest/user/file-size.html) page, it suggests doing the following:
```py
reader = PdfReader("example.pdf")
...

for page in reader.pages:
    page.compress_content_streams()  # This is CPU intensive!
    writer.add_page(page)
```

However, this can't work, as there is a [specific exception](https://github.com/py-pdf/pypdf/blob/534c7b4bf7052c103d94e5761631979e23411e39/pypdf/_page.py#L1787) ensuring that page is from `PdfWriter` is being used (code sample is using page from `PdfReader`).

---
We found this as we've recently updated from `~2.8.X` --> `3.0.X` and now our 600kb files from the deprecated `PdfMerger` are now **10mb (!)** files by using `PdfWriter`, so we desperately need to reduce file sizes, if you have any more tips about that :)

**Comment by stefan6419846** [comment]
You should use the correct RTD page, which seems to use the correct code: https://pypdf.readthedocs.io/en/latest/user/file-size.html#lossless-compression

**Comment by Dotrar** [comment]
Thanks for the clarification. quite a mix-up with the two projects.

**Comment by MartinThoma** [comment]
@Dotrar I agree that the read the docs (RTD) generated docs of PyPDF2 are confusing as they are still linked in a lot of places + look very similar to pypdf. I've opened https://github.com/py-pdf/pypdf/issues/2051 to fix that.

**Comment by Dotrar** [comment]
@MartinThoma thanks for that.

Regarding our issue at hand, we've found a massive increase in filesizes when joining multiple pdf's together after we've updated from the old pypdf2 `PdfMerger` to the newer pypdf `PdfWriter` - using the same example code on the docs.

Is there some documentation changes or something obvious we're missing or should i just make a new ticket?

ATM, the solution for us is to revert back and use to the older pyPDF2 ( ._.)
